### ISSUE TITLE
[RFC] Data Augmentation of object detection

### ISSUE THREAD

**Issue by zhiqwang** [Quick-AI/quickvision#83]
## 🚀 Feature

Hi, Recently I found a repo about data augmentation repo with excellent programming, named [TensorPipe](https://github.com/kartik4949/TensorPipe). But it's implemented using tensorflow, do you mind to add a similar implementation based in PyTorch in our quickvision repo?

### Motivation

Augmentation play an important role in practical application especially when the collection of images and its labels is restricted by various reasons or the cost is very high.

### Pitch

- [ ] Supports Mosiac Augmentation
- [ ] Supports  CutOut

**Comment by oke-aditya** [comment]
Yes we can have such augmentation. Especially bounding box augmentations. As of now torchvision itself is getting upgrade for transforms. So maybe after a few releases we will have this. 

We can plan for native ops module too.

**Comment by oke-aditya** [comment]
Once we have a nice nested tensor support in PyTorch as pointed out in torchvision `(vmap, etc)`
It wil be possible for torchvision itself to have params such as probablistic transforms, keep some batches from transforms.
This will make transforms even more powerful.

**Comment by zhiqwang** [comment]
That will be cool!
